soh/src/code/z_actor.c

```diff
@@ -2182,7 +2185,14 @@ void func_8002F7A0(PlayState* play, Actor* actor, f32 arg2, s16 arg3, f32 arg4)
 }
 
 void func_8002F7DC(Actor* actor, u16 sfxId) {
-    Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    if (actor->id != ACTOR_PLAYER || sfxId < NA_SE_VO_LI_SWORD_N || sfxId > NA_SE_VO_LI_ELECTRIC_SHOCK_LV_KID) {
+        Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &D_801333E0 , &D_801333E0, &D_801333E8);
+    }
+    else {
+        freqMultiplier = CVarGetFloat("gLinkVoiceFreqMultiplier", 1.0);
+        if (freqMultiplier <= 0) freqMultiplier = 1;
+        Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &freqMultiplier, &D_801333E0, &D_801333E8);
+    }
```

Comment:
```suggestion
    if (CVarGetInteger("gModifyVoiceFrequency", 0) &&
        actor->id == ACTOR_PLAYER &&
        sfxId >= NA_SE_VO_LI_SWORD_N &&
        sfxId <= NA_SE_VO_LI_ELECTRIC_SHOCK_LV_KID) {
        
        freqMultiplier = CVarGetFloat("gLinkVoiceFreqMultiplier", 1.0);
        Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &freqMultiplier, &D_801333E0, &D_801333E8);
        return;
    }

    Audio_PlaySoundGeneral(sfxId, &actor->projectedPos, 4, &D_801333E0 , &D_801333E0, &D_801333E8);
```

my main concern with how you have it now is that it's *always* using `freqMultiplier` for the link's voice. I think adding a checkbox to turn that on/off and being able to revert to fully authentic behavior would be ideal

Reply:
How about just reverting to authentic behavior if the cvar get returns `1.0`? This would avoid a seemingly redundant and unnecessary UI element for the checkbox, reducing UI clutter.

Reply:
just a general hesitation around doing a bunch of float comparisons